pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -107,8 +95,25 @@ type PluginManager struct {
 	kubeClient      pluginsCore.KubeClient
 	metrics         PluginMetrics
 	// Per namespace-resource
-	backOffController    *backoff.Controller
-	resourceLevelMonitor *ResourceLevelMonitor
+	backOffController     *backoff.Controller
+	resourceLevelMonitor  *ResourceLevelMonitor
+	ignoreOwnerReferences bool
```

Comment:
how about finalizer? Having a finalizer is fine is it?

Reply:
Hi @kumare3 sorry for the delay. Yea, I just double-checked this and it's better if we have an option to override the k8s config `inject-finalizer` property.

Updated the PR with it.